common/display/displayplanemanager.cpp

```diff
@@ -175,7 +175,30 @@ std::tuple<bool, DisplayPlaneStateList> DisplayPlaneManager::ValidateLayers(
   for (auto j = layers->rbegin(); j != layers->rend(); ++j) {
     if (j->GetBuffer()->GetUsage() & kLayerCursor) {
       cursor_layer = &(*(j));
-      layer_end = std::next(j).base();
+      // Handle Cursor layer.
+      DisplayPlane *cursor_plane = NULL;
+      if (cursor_layer) {
+        // Handle Cursor layer. If we have dedicated cursor plane, try using it
+        // to composite cursor layer.
+        if (cursor_plane_)
+          cursor_plane = cursor_plane_.get();
+        if (cursor_plane) {
+          commit_planes.emplace_back(OverlayPlane(cursor_plane, cursor_layer));
+          // Lets ensure we fall back to GPU composition in case
+          // cursor layer cannot be scanned out directly.
+          if (FallbacktoGPU(cursor_plane, cursor_layer, commit_planes)) {
+            cursor_plane = NULL;
+          }
+        }
+
+        // We need to do this here to avoid compositing cursor with any previous
```

Comment:
u  can remove these comments now. Can u test on SKL too before commiting the changes